README.md

```diff
@@ -157,37 +189,160 @@ private AuthenticationCallback getAuthInteractiveCallback() {
+
+```java
+ISingleAccountApplication mSingleAccountApp = null;
+IAccount mAccount;
+private static final List<String> SCOPES = Collections.singletonList("User.Read"); // Basic Microsoft Graph scope
 
+// Create a Single Account PublicClientApplication
+PublicClientApplication.createSingleAccountPublicClientApplication(getContext(),
+    R.raw.msal_config,
+    new IPublicClientApplication.ISingleAccountApplicationCreatedListener() {
+        @Override
+        public void onCreated(ISingleAccountPublicClientApplication application) {
+            mSingleAccountApp = application;
+        }
+
+        @Override
+        public void onError(MsalException exception) {
+            // Log and handle PCA creation failure
+        }
+    });
 ```
 
+```java
+// Sign In
+SignInParameters parameters = new SignInParameters.Builder()
+    .startActivity(activity)
+    .withCallback(getAuthInteractiveCallback())
+    .build();
+mSingleAccountApp.signIn(parameters);
+```
+
+```java
+// Get current signed-in account
+mSingleAccountApp.getCurrentAccountAsync(new ISingleAccountPublicClientApplication.CurrentAccountCallback() {
+    @Override
+    public void onAccountLoaded(@Nullable IAccount account) {
+        if (account != null) {
+            // Store the account for later use
+            mAccount = account;
+            // Account is signed in
+            String username = account.getUsername();
+        } else {
+            // No account is signed in
+        }
+    }
+
+    @Override
+    public void onAccountChanged(@Nullable IAccount priorAccount, @Nullable IAccount currentAccount) {
+        // Account has changed, update UI accordingly
+    }
+
+    @Override
+    public void onError(@NonNull MsalException exception) {
+        // Handle error loading account
+    }
+});
+```
+
+```java
+// Silent Token Acquisition
+AcquireTokenSilentParameters silentParameters = new AcquireTokenSilentParameters.Builder()
+    .withScopes(SCOPES)
+    .forAccount(account)
+    .withCallback(getAuthInteractiveCallback())
+    .build();
+mSingleAccountApp.acquireTokenSilentAsync(silentParameters);
+```
+
+```java
+// Silent Token Acquisition synchronously, must be done in a background thread.
+AcquireTokenSilentParameters silentParameters = new AcquireTokenSilentParameters.Builder()
+    .withScopes(SCOPES)
+    .forAccount(account)
+    .build();
+mSingleAccountApp.acquireTokenSilent(silentParameters);
+```
+
+All of the above examples can be found in the `snippets` directory, where we've also included jotlin examples.
+
+>**WARNING**: Device Code Flow (`acquireTokenWithDeviceCode`) is discouraged due to security concerns in the industry. Only use this authentication method in specific scenarios where devices lack input capabilities necessary for interactive authentication (e.g., conference room devices without text input). For standard Android applications, always use the interactive `acquireToken()` or `signIn()` APIs demonstrated above.
```

Comment:
"Only use this authentication method in specific scenarios where devices lack input capabilities necessary for interactive authentication (e.g., conference room devices without text input)."

should we even mentioned this?

Would it make more sense to say something like "we're keeping this for backcompat" (which is true, since DCF will soon be replaced by Resource account. @mohitc1 please correct me if i'm wrong)

Reply:
I think it's fine to mention this part. It's important to mention why the auth flow was enabled in the first place in my opinion. Once we've deprecated it and provided an alternative, i think we can skip mentioning it all together.